libres/lib/res_util/block_fs.cpp

```diff
@@ -93,20 +90,10 @@ struct block_fs_struct {
     int data_fd;
     FILE *data_stream;
 
-    long int data_file_size; /* The total number of bytes in the data_file. */
-    int block_size;          /* The size of blocks in bytes. */
-
     std::mutex mutex;
 
-    hash_type *
-        index; /* THE HASH table of all the nodes/files which have been stored. */
-    vector_type
-        *file_nodes; /* This vector owns all the file_node instances - the index
-                       structure only contain pointers to the objects stored in
-                       this vector. */
-    int write_count; /* This just counts the number of writes since the file system was mounted. */
+    std::unordered_map<std::string, BlockMeta> index;
```

Comment:
`index` is somewhat too short to understand what is going on. I guess its the metadata for each keyword? `keyword_meta`?